### PR TITLE
feat(types): accept read-only event names

### PR DESCRIPTION
**Pull request by jablko**
Add `readonly` modifier to event names parameters.

### CODE REVIEW

File: src/event-handler/on.ts [hunk 1]
```diff
@@ -19,10 +19,16 @@ function handleEventHandlers(
...
   state: State,
-  webhookNameOrNames: EmitterWebhookEventName | EmitterWebhookEventName[],
+  webhookNameOrNames:
+    | EmitterWebhookEventName
+    | readonly EmitterWebhookEventName[],
   handler: Function
 ) {
-  if (Array.isArray(webhookNameOrNames)) {
+  if (
+    (Array.isArray as (arg: any) => arg is readonly unknown[])(
```
Comment (G-Rath) [inline src/event-handler/on.ts:28]: Not a fan of this - I don't think `readonly` gives us much advantages to be worth needing to cast like this, nor to have extra runtime code.

Comment (jablko) [reply src/event-handler/on.ts:28]: What's the best workaround?

Comment (oscard0m) [reply src/event-handler/on.ts:28]: @G-Rath

Comment (oscard0m) [reply src/event-handler/on.ts:28]: @jablko is there a specific use case where you find these changes valuable? Maybe with the example you would like to get type-covered we can come up together with other alternatives.

Comment (G-Rath) [reply src/event-handler/on.ts:28]: > What's the best workaround?

There isn't really a workaround.

**Comment by gr2m** [comment]
I agree with @G-Rath's point here: https://github.com/octokit/webhooks.js/pull/455/files#r574053711. Is there any advantage of using a readonly array here? I don't think a problem ever came up from users that would be resolved by this change

**Comment by gr2m** [comment]
closing due to inactivity

**Pull request closed without merging**